src/main/java/uk/gov/laa/gpfd/model/MappingTable.java

```diff
@@ -1,22 +1,67 @@
 package uk.gov.laa.gpfd.model;
 
-//TODO - This class will be a bean, dont forget to annotate it with  @org.springframework.beans.factory.annotation.Autowired to make sonarlint happy
-// Bean guide: https://www.baeldung.com/spring-bean
+import java.util.Optional;
 
 /**
  * A class representing the data in the MOJFIN 'CSV - SQL Mapping' Table. A subset of this data will eventually be
  * returned to the user via the /reports endpoint, in the form of a ReportListResponse
  */
 public record MappingTable(
-        int id,
-        String reportName,
-        String excelReport,
-        String csvName,
-        int excelSheetNum,
-        String sqlQuery,
-        String baseUrl,
-        String reportOwner,
-        String reportCreator,
-        String description,
-        String ownerEmail
-) {}
+        Optional<Integer> id,
```

Comment:
Is there a reason these all need to be optional? Surely things like an id is rather essential?

Reply:
Thank you for the question!

The reason I moved to using explicitly `Optional` for all fields, including `id`, is to make explicit the structure of the data as it might be retrieved from the database, especially in cases where each column may be nullable. 

In the case of the `CSV_TO_SQL_MAPPING_TABLE`, table going to be created from: 
```
"GPFD"."CSV_TO_SQL_MAPPING_TABLE" 
   (	
        "ID" NUMBER(*,0), 
	"REPORT_NAME" VARCHAR2(200 BYTE), 
	"SQL_QUERY" VARCHAR2(4000 BYTE),
	"BASE_URL" VARCHAR2(450 BYTE), 
	"REPORT_OWNER" VARCHAR2(100 BYTE), 
	"REPORT_CREATOR" VARCHAR2(100 BYTE), 
	"REPORT_DESCRIPTION" VARCHAR2(400 BYTE), 
	"EXCEL_REPORT" VARCHAR2(200 BYTE), 
	"EXCEL_SHEET_NUM" NUMBER, 
	"CSV_NAME" VARCHAR2(200 BYTE), 
	"OWNER_EMAIL" VARCHAR2(300 BYTE)
   ) 
   ```
where fields like `id`,`reportName`, or `sqlQuery` might be missing or have null values, depending on the specific records we retrieve.

While it's true that in many contexts, an `id` is essential and typically expected, the use of `Optional` allows us to safely handle cases where data might be missing (defensive programming) without introducing `NullPointerExceptions`.

By using `Optional`, we explicitly express that certain fields might be absent and allow the calling code to handle those cases appropriately, making the model more flexible and robust when interacting with the database.